_posts/2019-04-10-lecture-23.md

```diff
@@ -0,0 +1,401 @@
+<figure>
+  <img src="{{ 'assets/img/notes/lecture-23/lda.png' | relative_url }}" />
+  <figcaption>
+    <strong>Figure 1.</strong>
+    Graphical representation of Latent Dirichlet Allocation <d-cite key="blei2003latent"></d-cite>
+  </figcaption>
+</figure>
+
+However, a naive implementation of Dirichlet process will fail in the case of topic models, because it cannot ensure the topics are shared. To be more specific, for a DP with a *continuous* base measure, we have zero probability of sampling the same atoms twice. As a result, a new set of topics will be generated every time we try to sample a document, making the topic model useless. Thus, we want the base measure to be discrete, but we don't want to pin down the number of possible topics as in the original LDA. In other words, we need an **infinite, discrete, random** base measure, which leads to the Hierarchical Dirichlet Process (HDP) model proposed by Teh et al. <d-cite key="teh2005sharing"></d-cite>
+
+### Hierarchical Dirichlet Process
+The key observation of Teh et al. <d-cite key="teh2005sharing"></d-cite> is that a sample from a Dirichlet process gives us exactly what we want -- an infinite, discrete, random base measure. Hence, we can formulate a *hierarchical* Dirichlet process, where
+<d-math block>
+\begin{aligned}
+G_0 \mid \gamma,H &\sim \mathrm{DP}(\gamma,H) \\
+G_j \mid \alpha_0,G_0 &\sim \mathrm{DP}(\alpha_0,G_0)
+\end{aligned}
+</d-math>
+Here, $G_0$ is the global measure distributed as a Dirichlet process with concentration parameter $\gamma$ and base probability measure $H$. $G_j$ are the random measures which are conditionally independent given $G_0$, with distributions given by a Dirichlet process with base probability measure $G_0$. We can use $G_j$ to describe the distribution of topics for each document. Figure 2 shows a simple illustration of the sampling process of a HDP.
+<figure>
+  <img src="{{ 'assets/img/notes/lecture-23/hdp.png' | relative_url }}" />
+  <figcaption>
+    <strong>Figure 2.</strong>
+    Sampling from a Hierarchical Dirichlet Process <d-cite key="teh2005sharing"></d-cite>
+  </figcaption>
+</figure>
+
+As in the case of DP, we can interpret HDP using the Chinese Restaurant Process analogy, where instead of a single restaurant, we now have a franchise of restaurants (documents). The restaurants share a common menu of dishes (topics). Each customer (word) picks a table in the restaurant and each table orders a dish (topic) according to the shared menu.
+
+The following process desribes how to sample from a HDP-LDA. For more details, please refer to the paper <d-cite key="teh2005sharing"></d-cite>.
+- Sample the global measure $G_0\sim\mathrm{DP}(\gamma,H)$ (e.g. using stick breaking)
+- For each document $j$
+    - Sample a distribution over topics $G_j\sim\mathrm{DP}(\alpha_0,G_0)$
+    - For $i=1,\dots,N_j$
```

Comment:
This equation doesn't render, probably need to use \ldots instead of \dots

Reply:
That's strange. It compiles on my local version. Changing to \ldots seems to fix it on the hosted version.